### PR TITLE
Fix the java doc contents in openJ9 class files to work with JDK file make/CompileJavaModules.gmk

### PR DESCRIPTION
**Pull request by archanaNogriya**
Fix the java doc contents in openJ9 class files to work with JDK file make/CompileJavaModules.gmk without making any patch changes.

Fixes : https://github.com/eclipse/openj9/issues/1569

Explanation:
Build has been failing with these 20 files.
**Basically these errors are related to 2 issues and the fixes is required to work is detailed here**

- code text in the generated HTML
Text within {@code ...} will be automatically HTML-escaped however <code> is not which is why above most of the files failed during compile.
{@code ...} is more concise: easier to read (and type).

The {@code} is equivalent to <code> except that it uses allows you to use angle brackets < and > instead of HTML-entities &lt; and &gt.
For example, {@code List<String>} is equivalent to  <code>List&lt;String&gt;</code>.

See also the documentation for {@code}.
https://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#%7B@code%7D

also

https://blogs.oracle.com/darcy/javadoc-tip:-code-and-literal

- Invalid syntax which has not been supported by HTML5
example : align or summary in is not supported in HTML5, 
another example.
also there were few error such as take not closed or parameter was not given.

These 20 files I have fixed and I have created a pull request.
After these fixes, My personal local build is successfully passed. 

Signed-off-by: Archana Nogriya <archana.nogriya@uk.ibm.com>

### CODE REVIEW

**Comment by archanaNogriya** [comment]
@DanHeidinga @pshipton

**Comment by archanaNogriya** [comment]
@DanHeidinga , 
I have made copyright fix adding 2018, still it fails for these files
17:28:42 The following files were modified and have incorrect copyrights
[Pipeline] echo
17:28:42 jcl/src/java.base/share/classes/java/lang/invoke/ConstantCallSite.java
[Pipeline] echo
17:28:42 jcl/src/java.base/share/classes/java/lang/invoke/MethodType.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/ClassLoadingMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/CompilationMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/GarbageCollectorMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/LockInfo.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/MemoryMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/MemoryManagerMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/MemoryPoolMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/OperatingSystemMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/PlatformLoggingMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/RuntimeMXBean.java
[Pipeline] echo
17:28:42 jcl/src/java.management/share/classes/java/lang/management/ThreadInfo.java

not sure what else I should be doing.

**Comment by keithc-ca** [comment]
If there are already two dates in the copyright notice, you should only ensure the second is the current year, not add a third. For example, in `ConstantCallSite.java` we want
```
Copyright (c) 2011, 2018 IBM Corp. and others
```
not
```
Copyright (c) 2011, 2011, 2018 IBM Corp. and others
```

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -2820,7 +2820,7 @@ public boolean isAnnotationPresent(Class<? extends Annotation> annotation) {
 
 /**
  * Cast this Class to a subclass of the specified Class. 
- * 
+ * @param <U> the type to cast this class object to
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/Class.java]: There shouldn't be two different @param statements for the single parameter.

File: jcl/src/java.base/share/classes/java/lang/String.java [hunk 2]
```diff
@@ -138,7 +138,7 @@ static void checkOffset(int offset, int length) {
 		 *			  an Object to compare
 		 * @param o2
 		 *			  an Object to compare
-		 * @return an int < 0 if object1 is less than object2, 0 if they are equal, and > 0 if object1 is greater
+		 * @return an int is negative 0 if object1 is less than object2, 0 if they are equal, and greater than 0 if object1 is greater
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/String.java]: "is negative" should be "less than". From "and greater than" remove "and"

File: jcl/src/java.base/share/classes/java/lang/StringBuffer.java [hunk 3]
```diff
@@ -316,8 +316,8 @@ public synchronized StringBuffer append (char[] chars) {
  * @param		length	the number of characters
  * @return		this StringBuffer
  *
- * @exception	IndexOutOfBoundsException when <code>length < 0, start < 0</code> or
- *				<code>start + length > chars.length</code>
+ * @exception	IndexOutOfBoundsException
+ *				If {@code length} is negative, {@code start} is negative or ({@code start} + {@code start}) is grater {@code chars.length}
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/StringBuffer.java]: `{@code start} + {@code start}` should be `{@code start + length}`

File: jcl/src/java.base/share/classes/java/lang/StringBuffer.java [hunk 4]
```diff
@@ -774,8 +774,8 @@ int capacityInternal() {
  * @param 		index	the zero-based index in this StringBuffer
  * @return		the character at the index
  *
- * @exception	IndexOutOfBoundsException when <code>index < 0</code> or
- *				<code>index >= length()</code>
+ * @exception	IndexOutOfBoundsException
+ *              If {@code index} is negative or {@code index} is grater or equal length()
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/StringBuffer.java]: typo in grater - greater

File: jcl/src/java.base/share/classes/java/lang/String.java (hunk 2 above)
Comment (archanaNogriya) [reply jcl/src/java.base/share/classes/java/lang/String.java]: Thanks @pshipton , I thought I have replaced this with {@code} .. let me see why my changes not reflected.

File: jcl/src/java.base/share/classes/java/lang/StringBuffer.java [hunk 5]
```diff
@@ -803,8 +803,8 @@ public synchronized char charAt(int index) {
  * @param		end	the offset one past the last character
  * @return		this StringBuffer
  *
- * @exception	StringIndexOutOfBoundsException when <code>start < 0, start > end</code> or
- *				<code>end > length()</code>
+ * @exception	StringIndexOutOfBoundsException
+ *				If {@code start} is negative, {@code start} is grater {@code end} or {@code end} is grater length()
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/StringBuffer.java]: grater - greater than (two places)

File: jcl/src/java.base/share/classes/java/lang/StringBuffer.java [hunk 6]
```diff
@@ -974,8 +974,10 @@ private void ensureCapacityImpl(int min) {
  * @param		buffer	the destination character array
  * @param		index	the starting offset in the character array
  *
- * @exception	IndexOutOfBoundsException when <code>start < 0, end > length(),
- *				start > end, index < 0, end - start > buffer.length - index</code>
+ * @exception	IndexOutOfBoundsException
+ *              If {@code start} is negative, {@code end} is greater length(),
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/StringBuffer.java]: greater - greater than

File: jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java [hunk 7]
```diff
@@ -3705,7 +3705,7 @@ private static boolean counterPredicate(int endValue, int counter) {
 	}
 	
 	/**
-	 * Produce a loop handle that iterates over a range of values produced by an Iterator<T>
+	 * Produce a loop handle that iterates over a range of values produced by an Iterator object
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java]: We want to keep the `<T>` generic.  Changing it to `Iterator object` changes the meaning.  Maybe wrap it in `{@code Iterator<T> }`

File: jcl/src/java.management/share/classes/java/lang/management/LockInfo.java [hunk 8]
```diff
@@ -145,7 +146,7 @@ public static LockInfo from(CompositeData compositeData) {
 	 * The string will hold both the name of the lock object's class and it's
 	 * identity hash code expressed as an unsigned hexadecimal. i.e.<br>
 	 * <p>
-	 * {@link #getClassName()}&nbsp;+&nbsp;&commat;&nbsp;+&nbsp;Integer.toHexString({@link #getIdentityHashCode()})
+	 * {@link #getClassName()} &nbsp;+&nbsp;&nbsp;+&nbsp;Integer.toHexString({@link #getIdentityHashCode()})
 	 * </p>
```
Comment (DanHeidinga) [inline jcl/src/java.management/share/classes/java/lang/management/LockInfo.java]: The `</p>` was deleted above.  Was that accidental or was this one missed?

File: jcl/src/java.base/share/classes/java/lang/String.java (hunk 2 above)
Comment (archanaNogriya) [reply jcl/src/java.base/share/classes/java/lang/String.java]: @pshipton Looks like you are seeing bit old code changes. Let me see why it is not refreshed with latest.

File: jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java (hunk 7 above)
Comment (archanaNogriya) [reply jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java]: Hi Dan, yes I have done all those changes.. some how it is not seen or refresh those latest changes here .. wired .. I am checking  on it. you will see latest one .. these are bit old changes then I made everything with {@code}

**Comment by pshipton** [comment]
Apparently the commit is out of date. Please update it and make a comment when its ready for review.

**Comment by pshipton** [comment]
Also, please improve the content of the commit comment. It should have some context, and mention the changes are to javadoc comments.

File: jcl/src/java.base/share/classes/java/lang/Class.java (hunk 1 above)
Comment (archanaNogriya) [reply jcl/src/java.base/share/classes/java/lang/Class.java]: Agree I will fix that. thanks

File: jcl/src/java.base/share/classes/java/lang/StringBuffer.java (hunk 4 above)
Comment (archanaNogriya) [reply jcl/src/java.base/share/classes/java/lang/StringBuffer.java]: Some how I missed this line.. actually this should be fixed with this line,
when {@code index < 0} or {@code index >= length()}

File: jcl/src/java.base/share/classes/java/lang/StringBuffer.java (hunk 5 above)
Comment (archanaNogriya) [reply jcl/src/java.base/share/classes/java/lang/StringBuffer.java]: Some how I missed this line.. actually this should be fixed with this line,
when {@code}

File: jcl/src/java.management/share/classes/java/lang/management/LockInfo.java (hunk 8 above)
Comment (archanaNogriya) [reply jcl/src/java.management/share/classes/java/lang/management/LockInfo.java]: In these cases build has been complaining  about close <p> tag unnecessary, while getting to understand why it should be a problem then found HTML never required the <p> tag to be closed—it was always optional. You may close your HTML tags so it looks like well-formed XHTML, but that isn't necessary. XHTML is more strict than HTML. and when I removed build was fine.

**Comment by archanaNogriya** [comment]
@pshipton @DanHeidinga , this commit is ready to review now, Code is uptodate now with all my changes.

**Comment by archanaNogriya** [comment]
@andrew-m-leonard

File: jcl/src/java.base/share/classes/java/lang/StringBuffer.java [hunk 9]
```diff
@@ -774,8 +774,8 @@ int capacityInternal() {
  * @param 		index	the zero-based index in this StringBuffer
  * @return		the character at the index
  *
- * @exception	IndexOutOfBoundsException when <code>index < 0</code> or
- *				<code>index >= length()</code>
+ * @exception	IndexOutOfBoundsException
+ *              If {@code index<0} or {@code index >= length()}
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/StringBuffer.java]: Please add spaces for `index < 0`

File: jcl/src/java.management/share/classes/java/lang/management/ClassLoadingMXBean.java [hunk 10]
```diff
@@ -30,18 +30,17 @@
...
+ * Accessing this <code>MXBean</code> can be done in one of three ways.
  * <ol>
- * <li>Invoking the static ManagementFactory.getClassLoadingMXBean() method.
- * </li>
+ * <li>Invoking the static ManagementFactory.getClassLoadingMXBean() method.</li>
  * <li>Using a javax.management.MBeanServerConnection.</li>
  * <li>Obtaining a proxy MXBean from the static
  * ManagementFactory.newPlatformMXBeanProxy(MBeanServerConnection connection,
- * String mxbeanName, Class <T>mxbeanInterface) method, passing in
+ * String mxbeanName, {@code Class<T> mxbeanInterface())} method, passing in
```
Comment (pshipton) [inline jcl/src/java.management/share/classes/java/lang/management/ClassLoadingMXBean.java]: Seems `@code` should be around the entire API and not just the single parameter

File: jcl/src/java.management/share/classes/java/lang/management/LockInfo.java [hunk 11]
```diff
@@ -145,7 +146,7 @@ public static LockInfo from(CompositeData compositeData) {
 	 * The string will hold both the name of the lock object's class and it's
 	 * identity hash code expressed as an unsigned hexadecimal. i.e.<br>
 	 * <p>
-	 * {@link #getClassName()}&nbsp;+&nbsp;&commat;&nbsp;+&nbsp;Integer.toHexString({@link #getIdentityHashCode()})
+	 * {@link #getClassName()} &nbsp;+&nbsp;&nbsp;+&nbsp;Integer.toHexString({@link #getIdentityHashCode()})
```
Comment (pshipton) [inline jcl/src/java.management/share/classes/java/lang/management/LockInfo.java]: Was it intentional to remove the `&comma`?

**Comment by pshipton** [comment]
Since this is ready to review, please remove the WIP from the title.

**Comment by archanaNogriya** [comment]
@pshipton : I was on vacation for last week, Let me make those changes as you suggested. thanks

File: jcl/src/java.management/share/classes/java/lang/management/LockInfo.java (hunk 11 above)
Comment (archanaNogriya) [reply jcl/src/java.management/share/classes/java/lang/management/LockInfo.java]: No, have not, let me try this one replacing. thanks

Comment (archanaNogriya) [reply jcl/src/java.management/share/classes/java/lang/management/LockInfo.java]: @keithc-ca  thanks it worked :)

**Pull request merged**